controllers/bookControllers.js

```diff
@@ -1,13 +1,44 @@
 const Book = require('../models/Book');
+const User = require('../models/User');
 const customError = require('../helpers/customErrorHandler');
 const mongoose = require('mongoose');
 
 exports.getBook = async (req, res, next) => {
   const { id } = req.params;
-
   try {
     let book = await Book.findById(id);
+    // id is a valid mongoose id
+    if (!book) {
+      next(customError(`Book with ID: ${id} does not exist`, 400));
+    }
     res.json(book);
+  } catch (err) {
+    // id is not a valid mongoose id
+    if (err instanceof mongoose.Error.CastError) {
+      next(customError(`ID: ${id} is not valid`, 400));
+    }
+    // server error
+    next(err);
+  }
+};
+
+exports.getUserLibrary = async (req, res, next) => {
+  const { city } = req.params;
+  // todo : check if its an actual city and make sure it starts with a capital letter
+  // todo : check the user has not matched or is not already an interest user of one of the books
+
+  try {
+    let userLibrary = await Book.find().where('city').equals(city);
+    if (userLibrary.length === 0) {
+      next(
+        customError(
+          `There are no book available to trade in this city: ${city}`,
+          400
+        )
+      );
+      return;
```

Comment:
do we need  a return if we call next?

Reply:
yes because if not the code will go through to the res and send back an empty array